packages/component_code_gen/templates/generate_polling_sources.py

```diff
@@ -0,0 +1,482 @@
+
+The app can also be an OAuth app. For OAuth integrations, this object exposes the OAuth access token in the variable `this.the_app_name_slug.$auth.oauth_access_token`. When you make the API request, make sure to use the format from the app docs, e.g. you may need to pass the OAuth access token as a Bearer token in the Authorization header.
+
+The object _may_ contain an optional a `props` property, which in the example below defines a string prop. The props object is not required. Include it only if the code connects to a Pipedream integration, or the code in the run method requires input. Props lets the user pass data to the step via a form in the Pipedream UI, so they can fill in the values of the variables. Include any required parameters as properties of the `props` object. Props must include a human-readable `label` and a `type` (one of string|boolean|integer|object) that corresponds to the Node.js type of the required param. string, boolean, and integer props allow for arrays of input, and the array types are "string[]", "boolean[]", and "integer[]" respectively. Complex props (like arrays of objects) can be passed as string[] props, and each item of the array can be parsed as JSON. If the user asks you to provide an array of object, ALWAYS provide a `type` of string[]. Optionally, props can have a human-readable `description` describing the param. Optional parameters that correspond to the test code should be declared with `optional: true`. Recall that props may contain an `options` method.
+
+Within the component's run method, the `this` variable refers to properties of the component. All props are exposed at `this.<name of the key in the props object>`. e.g. `this.input`. `this` doesn't contain any other properties.
+
+The run method is called when the component receives an event. The event is passed as the first and only argument to the run method. The event is a JSON object that contains the data from the webhook. The event is emitted by calling `this.$emit`. The first argument to `$emit` is the data to emit. You should only pass relevant data. For example, usually only the event.body is relevant. Headers and others are used to validate the webhook, but shouldn't be emitted. The second argument is an object that contains three fields: `id`, `summary`, and `ts`. The `id` field is a unique identifier for the event. The `summary` field is a human-readable summary of the event. The `ts` field is a timestamp of the event.
+
+There are also two other props in sources: `http` and `db`. `http` is an interface that lets you receive and respond to HTTP requests. `db` is a data store that lets you store data between runs of the component. You should always include both.
```

Comment:
`db` is a data store that lets you store data between runs of the component -> you should clarify that `db` is a type of prop `$.service.db` and explain that API in detail. Maybe in the rules section above.